File: services/servico-accounting/src/application/ports/EventPublisher.ts
import { DomainEvent } from '../../domain/events/DomainEvent';

/**
 * Porta de saída para publicação de eventos de domínio. Em produção,
 * corresponde ao tópico Kafka onde `ServicoAccounting` publica
 * `CreditReserved`/`CreditRejected` (ver EC-M5.2). Nesta primeira
 * versão, um EventBus em memória cumpre esse papel.
 */
export interface EventPublisher {
  publish(event: DomainEvent): Promise<void>;
}

File: services-java/servico-order/src/main/java/br/com/ftgo/order/application/port/EventPublisher.java
package br.com.ftgo.order.application.port;

import br.com.ftgo.order.domain.event.DomainEvent;

/**
 * Porta de saída para publicação de eventos de domínio. Em produção,
 * corresponde ao tópico Kafka {@code order_events} (ver EC-M5.2). Nesta
 * primeira versão, um publisher em memória cumpre esse papel e também
 * permite inspecionar os eventos publicados durante testes e a
 * demonstração.
 */
public interface EventPublisher {

    void publish(DomainEvent event);
}
